components/org.wso2.carbon.identity.api.server.script.library/org.wso2.carbon.identity.api.server.script.library.v1/src/main/java/org/wso2/carbon/identity/api/server/script/library/v1/impl/ScriptLibrariesApiServiceImpl.java

```diff
@@ -45,9 +52,17 @@ public Response addScriptLibrary(String name, InputStream contentInputStream, At
                                      String description) {
 
         serverScriptLibrariesService.addScriptLibrary(name, contentInputStream, description);
-        URI location =
-                ContextLoader.buildURIForHeader(V1_API_PATH_COMPONENT + SCRIPT_LIBRARY_PATH_COMPONENT + "/" + name);
-        return Response.created(location).build();
+        try {
+            URI location =
+                    ContextLoader.buildURIForHeader(V1_API_PATH_COMPONENT + SCRIPT_LIBRARY_PATH_COMPONENT
+                            + "/" + URLEncoder.encode(name, StandardCharsets.UTF_8.name())
+                            .replace("+", "%20"));
+            return Response.created(location).build();
+        } catch (UnsupportedEncodingException e) {
+            FunctionLibraryManagementException error = new FunctionLibraryManagementException(
+                    Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL.getMessage(), e);
+            throw handleScriptLibraryError(error, Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL);
```

Comment:
ErrorResponse errorResponse =
                new ErrorResponse.Builder().withCode(Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL.getCode()).withMessage(Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL.getMessage()).withDescription(Constants.ErrorMessage.ERROR_CODE_ERROR_ENCODING_URL.getDescription()).build();
        throw new APIError(Response.Status.INTERNAL_SERVER_ERROR, errorResponse);

Reply:
Fixed in f615b5d